Std/Data/BitVec/Bitblast.lean

```diff
@@ -18,11 +18,8 @@ The module is named for the bit-blasting operation in an SMT solver that convert
 expressions into expressions about individual bits in each vector.
 
 ## Main results
-* `x + y : BitVec w` is equivalent to `adc x y false`.
+* `x + y : BitVec w` is `(adc x y false).2`.
```

Comment:
I'm still not thrilled by this file, but not putting my finger on it.

My main complaint is that it proves that something I already knew about is equal to something new.

(Does that make sense? The new thing needs to be introduced and motivated first, and then its interesting to prove old and new things are equal. Sorry if that is a bit vague.)

Reply:
I can understand your point. In some sense, `adc` is an alternative definition of the addition operation we already have. But, having `adc` allows us to prove things about addition from the perspective as a bit-by-bit transformation with one bit of state.

This was helpful, for example, when proving `x + ~~~x = -1#w`. The argument here is that, actually, the carry bit is always false, and the proof went as follows:
```lean
rw [add_as_adc, adc, iunfoldr_replace (fun _ => false) (-1#w)]
  · rfl
  · simp [adcb]
```

This doesn't necessarily mean we should keep exposing `adc`. Probably we should find the fundamental theorems about `add` that would allow us to make this argument, and expose only those. What comes to mind is:
* Some theorem about `(concat xs x) + (concat ys y) = concat (xs + ys + _) _`
* An analogue of `iunfoldr_replace`, essentially a coinduction principle for addition as an arbitrary-width bitvector transformation

Reply:
Thanks for the explanations.